src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java

```diff
@@ -0,0 +1,578 @@
+                for (String type : dependencies.keySet()) {
+                    dependencyEntities.addAll(createDependencyEntities(site, path, dependencies.get(type), type, toRet));
+                }
+            }
+        }
+        logger.debug("Preparing transaction for database updates.");
+        DefaultTransactionDefinition defaultTransactionDefinition = new DefaultTransactionDefinition();
+        defaultTransactionDefinition.setName("upsertDependencies");
+        logger.debug("Starting transaction.");
+        TransactionStatus txStatus = transactionManager.getTransaction(defaultTransactionDefinition);
+        try {
+            logger.debug("Delete all source dependencies for list of paths site: " + site);
+            for (String path : paths) {
+                deleteAllSourceDependencies(site, path);
+            }
+            logger.debug("Insert all extracted dependencies entries lof list of paths for site: " + site);
+            insertDependencies(dependencyEntities);
+            logger.debug("Committing transaction.");
+            transactionManager.commit(txStatus);
+        } catch (Exception e) {
+            logger.debug("Rolling back transaction.");
+            transactionManager.rollback(txStatus);
+            throw new ServiceException("Failed to upsert dependencies for site: " + site + " paths: " + sbPaths.toString(), e);
+        }
+
+        return toRet;
+    }
+
+    private void deleteAllSourceDependencies(String site, String path) {
+        logger.debug("Delete all source dependencies for site: " + site + " path: " + path);
+        Map<String, String> params = new HashMap<String, String>();
+        params.put("site", site);
+        params.put("path", path);
+        dependencyMapper.deleteAllSourceDependencies(params);
+    }
+
+    private List<DependencyEntity> createDependencyEntities(String site, String path, Set<String> dependencyPaths, String dependencyType, Set<String> extractedPaths) {
+        logger.debug("Create dependency entity TO for site: " + site + " path: " + path);
+        List<DependencyEntity> dependencyEntities = new ArrayList<>();
+        if (dependencyPaths != null && dependencyPaths.size() > 0) {
+            for (String file : dependencyPaths) {
+                DependencyEntity dependencyObj = new DependencyEntity();
+                dependencyObj.setSite(site);
+                dependencyObj.setSourcePath(getCleanPath(path));
+                dependencyObj.setTargetPath(getCleanPath(file));
+                dependencyObj.setType(dependencyType);
+                dependencyEntities.add(dependencyObj);
+                extractedPaths.add(file);
+            }
+        }
+        return dependencyEntities;
+    }
+
+    private String getCleanPath(String path) {
+        return path.replaceAll("//", "/");
+    }
+
+    private void insertDependencies(List<DependencyEntity> dependencyEntities) {
```

Comment:
Rename to insertDependenciesIntoDatabase? (in our continued mission to make clearer method names...)